cli/src/pcluster/config/cluster_config.py

```diff
@@ -1851,6 +1851,25 @@ def _register_validators(self):
         self._register_validator(
             DuplicateNameValidator, name_list=[queue.name for queue in self.queues], resource_name="Queue"
         )
+        self._register_validator(
+            MaxCountValidator,
+            resources_length=len(self.queues),
+            max_length=self.settings.scheduler_definition.requirements.queue_constraints.max_count
+            if self.settings.scheduler_definition.requirements
+            and self.settings.scheduler_definition.requirements.queue_constraints
+            else MAX_NUMBER_OF_QUEUES,
```

Comment:
nit: you could use the utility get_attr from utils.py: `get_attr(self.settings.scheduler_definition, "equirements.queue_constraints.max_count", default=MAX_NUMBER_OF_QUEUES)`. Same below

Reply:
Good idea! Changed